src/Rpc/NodeGuardService.cs

```diff
@@ -859,4 +864,30 @@ public override async Task<GetWithdrawalsRequestStatusResponse> GetWithdrawalsRe
             }
         };
     }
+
+    public override async Task<GetChannelResponse> GetChannel(GetChannelRequest request, ServerCallContext context)
+    {
+        var channel = await _channelRepository.GetById(request.ChannelId);
+        if (channel == null)
+        {
+            throw new RpcException(new Status(StatusCode.NotFound, "Channel not found"));
+        }
+        
+        var result = new GetChannelResponse()
+        {
+            FundingTx = channel.FundingTx,
+            OutputIndex = channel.FundingTxOutputIndex,
+            ChanId = channel.ChanId,
+            SatsAmount = channel.SatsAmount,
+            Status = (CHANNEL_STATUS)((int)channel.Status - 1),
```

Comment:
I would do a direct matching between channel statuses and the possible statuses of `GetChannelResponse`. This line is not readable at all and it miss the point of using enums